src/model/interactions.rs

```diff
@@ -0,0 +1,108 @@
+//! Interactions information-related models
+
+use super::prelude::*;
+use crate::internal::prelude::*;
+
+/// Information about an Interaction
+///
+/// An interaction is the base "thing" that is sent when a user invokes a command, and is the same
+/// for Slash Commands and other future interaction types.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[non_exhaustive]
+pub struct Interaction {
+    pub id: InteractionId,
+    #[serde(rename = "type")]
+    pub kind: InteractionType,
+    pub data: Option<ApplicationCommandInteractionData>,
+    pub guild_id: GuildId,
+    pub channel_id: ChannelId,
+    pub member: Member,
+    pub token: String,
+    pub version: i32
```

Comment:
```suggestion
    pub version: i32,
```
Missing trailing comma.
Versions can be negative?

Reply:
oh yeah, good point, especially as it currently will always be 1 according to the docs, this should probably be an u8 or at least an u32